### PR TITLE
Adding support for lowercasing only the first character (#59)

### PR DESCRIPTION
**Pull request by fnandocontreras**

### CODE REVIEW

File: EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using System.Globalization;
+
+namespace EFCore.NamingConventions.Internal
+{
+    public class LowerCaseFirstCharacterNameRewriter : INameRewriter
+    {
+        private readonly CultureInfo _culture;
+
+        public LowerCaseFirstCharacterNameRewriter(CultureInfo culture) => _culture = culture;
+        public string RewriteName(string name)
```
Comment (roji) [inline EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs:10]: nit: can you please rewrite this method as an expression-bodied method (with the ternary operator)?

File: EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using System.Globalization;
+
+namespace EFCore.NamingConventions.Internal
+{
+    public class LowerCaseFirstCharacterNameRewriter : INameRewriter
+    {
+        private readonly CultureInfo _culture;
+
+        public LowerCaseFirstCharacterNameRewriter(CultureInfo culture) => _culture = culture;
```
Comment (roji) [inline EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs:9]: nit: insert empty line between the constructor and the method

File: EFCore.NamingConventions/Internal/NamingConvention.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@ public enum NamingConvention
         None,
         SnakeCase,
         LowerCase,
+        LowerCaseFirstCharacter,
```
Comment (roji) [inline EFCore.NamingConventions/Internal/NamingConvention.cs:8]: We can just call this CamelCase, no?

File: EFCore.NamingConventions/NamingConventionsExtensions.cs [hunk 4]
```diff
@@ -83,5 +83,24 @@ public static class NamingConventionsExtensions
             [NotNull] this DbContextOptionsBuilder<TContext> optionsBuilder, CultureInfo culture = null)
             where TContext : DbContext
             => (DbContextOptionsBuilder<TContext>)UseUpperSnakeCaseNamingConvention((DbContextOptionsBuilder)optionsBuilder, culture);
+
+        public static DbContextOptionsBuilder UseLowerCaseFirstCharacterNamingConvention(
```
Comment (roji) [inline EFCore.NamingConventions/NamingConventionsExtensions.cs:87]: UseCamelCaseNamingConvention

**Review by roji — CHANGES_REQUESTED** [review]
Thanks! Looks good overall, see some nits below.

File: EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs (hunk 1 above)
Comment (fnandocontreras) [reply EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs:10]: ok

File: EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs (hunk 2 above)
Comment (fnandocontreras) [reply EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs:9]: ok

File: EFCore.NamingConventions/Internal/NamingConvention.cs (hunk 3 above)
Comment (fnandocontreras) [reply EFCore.NamingConventions/Internal/NamingConvention.cs:8]: ok

File: EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs (hunk 1 above)
Comment (fnandocontreras) [reply EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs]: done

File: EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs (hunk 2 above)
Comment (fnandocontreras) [reply EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs:9]: done

File: EFCore.NamingConventions/Internal/NamingConvention.cs (hunk 3 above)
Comment (fnandocontreras) [reply EFCore.NamingConventions/Internal/NamingConvention.cs]: done

File: EFCore.NamingConventions/NamingConventionsExtensions.cs (hunk 4 above)
Comment (fnandocontreras) [reply EFCore.NamingConventions/NamingConventionsExtensions.cs]: done

File: README.md [hunk 5]
```diff
@@ -44,6 +44,7 @@ SELECT c.id, c.full_name
 
 * UseSnakeCaseNamingConvention: `FullName` becomes `full_name`
 * UseLowerCaseNamingConvention: `FullName` becomes `fullname`
+* UseLowerCaseFirstCharacterNamingConvention: `FullName` becomes `fullName`
```
Comment (roji) [inline README.md:47]: And here

File: EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+using System.Globalization;
+
+namespace EFCore.NamingConventions.Internal
+{
+    public class LowerCaseFirstCharacterNameRewriter : INameRewriter
```
Comment (roji) [inline EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs:5]: Also need to rename to CamelCaseNameRewriter

**Review by roji — COMMENTED** [review]
Just a few more renames and this looks ready.

File: README.md (hunk 5 above)
Comment (fnandocontreras) [reply README.md:47]: done

File: EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs (hunk 6 above)
Comment (fnandocontreras) [reply EFCore.NamingConventions/Internal/LowerCaseFirstCharacterNameRewriter.cs]: done, alsi renamed some others remaining, (see in last commit)

**Pull request merged**